### PR TITLE
[core] remove import thread

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

import thread is no longer needed, remove it.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scv119** [comment]
@iycheng do we even need function_manager?

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
     from typing_extensions import Literal, Protocol
 
 import ray
-import ray._private.import_thread as import_thread
+import ray._private.gcs_utils as gcs_utils
```
Comment (pcmoritz) [inline python/ray/_private/worker.py:46]: This is not allowed (there is a check that will fail) as it brings grpc back into the `import ray` critical path :)

But you don't need this, right?

**Comment by pcmoritz** [comment]
There is also `join_import_thread` in worker.py and `def start_import_thread():` that should be removed as well I think :)

Thanks a lot for doing this!!!

**Comment by iycheng** [comment]
@scv119 function manager is still needed I think. Doc [related](https://docs.google.com/document/d/1yYSDuuek_IvZ-cR7HOHzNy_Yn7LIHAka_zpPxzx69oo/edit#heading=h.vc1ayvdmw1n8) to this.

Btw, can you confirm that `run_functions_on_all_workers` have been deprecated? Removing this function will lead that code path failed.

**Comment by scv119** [comment]
> Btw, can you confirm that run_functions_on_all_workers have been deprecated? Removing this function will lead that code path failed.

yup `run_functions_on_all_workers` is gone in the codebase.

**Review by fishbone — APPROVED** [review]
LGTM

**Pull request merged**